src/Planners/src/QuinticSpline.cpp

```diff
@@ -0,0 +1,787 @@
+                  << knotSize << ", Retrieved: " << initialCondition.velocity.size() << "."
+                  << std::endl;
+        return false;
+    }
+
+    if (knotSize != initialCondition.acceleration.size())
+    {
+        std::cerr << "[QuinticSpline::Impl::setBoundaryVelocitiesAndAccelaration] The size of the "
+                     "initial acceleration is different from the expected one. Expected: "
+                  << knotSize << ", Retrieved: " << initialCondition.acceleration.size() << "."
+                  << std::endl;
+        return false;
+    }
+
+    if (knotSize != finalCondition.velocity.size())
+    {
+        std::cerr << "[QuinticSpline::setBoundaryVelocitiesAndAccelaration] The size of the "
+                     "final velocity is different from the expected one. Expected: "
+                  << knotSize << ", Retrieved: " << finalCondition.velocity.size() << "."
+                  << std::endl;
+        return false;
+    }
+
+    if (knotSize != finalCondition.acceleration.size())
+    {
+        std::cerr << "[QuinticSpline::setBoundaryVelocitiesAndAccelaration] The size of the "
+                     "final acceleration is different from the expected one. Expected: "
+                  << knotSize << ", Retrieved: " << finalCondition.acceleration.size() << "."
+                  << std::endl;
+        return false;
+    }
+
+    // store the boundary conditions
+    knots.front().velocity = initialCondition.velocity;
+    knots.front().acceleration = initialCondition.acceleration;
+
+    knots.back().velocity = finalCondition.velocity;
+    knots.back().acceleration = finalCondition.acceleration;
+
+    return true;
+}
+
+void QuinticSpline::Impl::addTripletCurrentKnot(const int& knotIndex,
+                                                const int& rowOffset,
+                                                const int& columnOffset,
+                                                std::vector<Eigen::Triplet<double>>& tripletList)
+{
+    const auto& poly = polynomials;
+
+    // The following triplets represent this matrix
+    // /    /   1      1\          /   1      1\ \
+    // | 12 |------ - --|      -3  |------ + --| |
+    // |    | 2        2|          |T        T | |
+    // |    |T        T |          \ i - 1    i/ |
+    // |    \ i - 1    i/                        |
+    // |                                         |
+    // |                                         |
+    // |    /   1      1\         / 1      1  \  |
+    // | 16 |------ + --|       3 |-- - ------|  |
+    // |    | 3        3|         | 2    2    |  |
+    // |    |T        T |         |T    T     |  |
+    // \    \ i - 1    i/         \ i    i - 1/  /
+
+    tripletList.push_back({rowOffset,
+                           columnOffset,
+                           12 * (1 / std::pow(poly[knotIndex - 1].duration, 2)
+                                 - 1 / std::pow(poly[knotIndex].duration, 2))});
```

Comment:
```suggestion
    tripletList.push_back({rowOffset,
                           columnOffset,
                           12 * ( (1 / std::pow(poly[knotIndex - 1].duration, 2)
                                 - (1 / std::pow(poly[knotIndex].duration, 2)) ) });
```

I'd more careful and explicit with the use of braces here. Similarly for other triplet values.

Reply:
To be more readable, get a const ref on `poly[knotIndex - 1]` and `poly[knotIndex]` maybe?

Reply:
I improved the readability in 4b2c8c02d83ba54b7eb80957caa042321ba16ee2